graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java

```diff
@@ -204,6 +214,16 @@ private DirectIndexQueryResult performAsyncDirectIndexQuery(SearchParams searchP
         }catch (Exception e) {
             LOG.error("Failed to execute direct query on ES {}", e.getMessage());
             throw new AtlasBaseException(AtlasErrorCode.INDEX_SEARCH_FAILED, e.getMessage());
+        } finally {
+            if (contextIdExists) {
+                // If the search context id is present, then we need to remove the search context from the cache
+                try {
```

Comment:
Can we double check this logic? If an exception is thrown at any point with the `try` block, should we always try to remove search context id from cache?

Reply:
Its a worst case scenario and if we don't remove it just cancellation won't work